### DISCUSSION TITLE
How to set z-index on FloatingPortal

### DISCUSSION THREAD

**Discussion by rob-johansen** [floating-ui/floating-ui#2640] [General]
I have two different components that use `<FloatingPortal>`. One is a drawer, which opens from the side of the screen. The other is a modal, which must be above everything else in terms of stacking context—i.e. if the drawer and modal are both open, the modal should be "on top" of the drawer.

My problem is that each `<FloatingPortal>` instance appears to produce its own `<div>`, and I can't get the modal's `<div>` to render after the drawer's `<div>` (and there doesn't seem to be a way to set `z-index` on a `<FloatingPortal>` instance):

```html
<body>
  <!--The modal's floating portal always comes first, for some reason.-->
  <div id=":Rbacm:" data-floating-ui-portal="" data-aria-hidden="true" aria-hidden="true">
  ...
  <!--The drawer's floating portal always comes later, for some reason.-->
  <div id=":r2:" data-floating-ui-portal="" data-aria-hidden="true" aria-hidden="true">
</body>
```

Any suggestions?

**Comment by atomiks** [comment]
If you're conditionally rendering the `FloatingPortal` component based on the open state, it will always be injected after the previous one and thus overlay on top of previous portals without needing to consider `z-index`. Are you always rendering the portal or conditionally rendering it?

**Comment by rob-johansen** [comment]
Ah, that seems to be my problem. I'm always rendering both, then using a global function to show/hide them anywhere in my app when needed. Looks like I've got some refactoring to do, but thanks for your help!
